### PR TITLE
Fix numpy 2 warnings emitted from skymatch

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR fixes warnings emitted by skymatch when running with numpy 2. These seem to be the majority of warnings emitted by numpy 2.

## Tasks
- [x] **request a review from someone specific**, to avoid making the maintainers review every PR
- [x] add a build milestone, i.e. `Build 11.3` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types) 
  - [ ] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
- ``changes/<PR#>.docs.rst``
- ``changes/<PR#>.stpipe.rst``
- ``changes/<PR#>.datamodels.rst``
- ``changes/<PR#>.scripts.rst``
- ``changes/<PR#>.fits_generator.rst``
- ``changes/<PR#>.set_telescope_pointing.rst``
- ``changes/<PR#>.pipeline.rst``

## stage 1
- ``changes/<PR#>.group_scale.rst``
- ``changes/<PR#>.dq_init.rst``
- ``changes/<PR#>.emicorr.rst``
- ``changes/<PR#>.saturation.rst``
- ``changes/<PR#>.ipc.rst``
- ``changes/<PR#>.firstframe.rst``
- ``changes/<PR#>.lastframe.rst``
- ``changes/<PR#>.reset.rst``
- ``changes/<PR#>.superbias.rst``
- ``changes/<PR#>.refpix.rst``
- ``changes/<PR#>.linearity.rst``
- ``changes/<PR#>.rscd.rst``
- ``changes/<PR#>.persistence.rst``
- ``changes/<PR#>.dark_current.rst``
- ``changes/<PR#>.charge_migration.rst``
- ``changes/<PR#>.jump.rst``
- ``changes/<PR#>.clean_flicker_noise.rst``
- ``changes/<PR#>.ramp_fitting.rst``
- ``changes/<PR#>.gain_scale.rst``

## stage 2
- ``changes/<PR#>.assign_wcs.rst``
- ``changes/<PR#>.badpix_selfcal.rst``
- ``changes/<PR#>.msaflagopen.rst``
- ``changes/<PR#>.nsclean.rst``
- ``changes/<PR#>.imprint.rst``
- ``changes/<PR#>.background.rst``
- ``changes/<PR#>.extract_2d.rst``
- ``changes/<PR#>.master_background.rst``
- ``changes/<PR#>.wavecorr.rst``
- ``changes/<PR#>.srctype.rst``
- ``changes/<PR#>.straylight.rst``
- ``changes/<PR#>.wfss_contam.rst``
- ``changes/<PR#>.flatfield.rst``
- ``changes/<PR#>.fringe.rst``
- ``changes/<PR#>.pathloss.rst``
- ``changes/<PR#>.barshadow.rst``
- ``changes/<PR#>.photom.rst``
- ``changes/<PR#>.pixel_replace.rst``
- ``changes/<PR#>.resample_spec.rst``
- ``changes/<PR#>.residual_fringe.rst``
- ``changes/<PR#>.cube_build.rst``
- ``changes/<PR#>.extract_1d.rst``
- ``changes/<PR#>.resample.rst``

## stage 3
- ``changes/<PR#>.assign_mtwcs.rst``
- ``changes/<PR#>.mrs_imatch.rst``
- ``changes/<PR#>.tweakreg.rst``
- ``changes/<PR#>.skymatch.rst``
- ``changes/<PR#>.exp_to_source.rst``
- ``changes/<PR#>.outlier_detection.rst``
- ``changes/<PR#>.tso_photometry.rst``
- ``changes/<PR#>.stack_refs.rst``
- ``changes/<PR#>.align_refs.rst``
- ``changes/<PR#>.klip.rst``
- ``changes/<PR#>.spectral_leak.rst``
- ``changes/<PR#>.source_catalog.rst``
- ``changes/<PR#>.combine_1d.rst``
- ``changes/<PR#>.ami.rst``

## other
- ``changes/<PR#>.wfs_combine.rst``
- ``changes/<PR#>.white_light.rst``
- ``changes/<PR#>.cube_skymatch.rst``
- ``changes/<PR#>.engdb_tools.rst``
- ``changes/<PR#>.guider_cds.rst``
</details>

### CODE REVIEW

File: jwst/skymatch/region.py [hunk 1]
```diff
@@ -413,18 +413,25 @@ def intersection(self, edge):
...
         v = edge._stop - edge._start
         w = self._start - edge._start
-        D = np.cross(u, v)
 
-        if np.allclose(np.cross(u, v), 0, rtol=0,
+        # Find the determinant of the matrix formed by the vectors u and v
+        #    Note: Originally this was computed using a numpy "2D" cross product,
+        #          however, this functionality has been deprecated and slated for
+        #          removal.
+        D = np.linalg.det([u, v])
```
Comment (WilliamJamieson) [inline jwst/skymatch/region.py:421]: This is numerically faster, but has a small numerical difference from using the `np.cross`. In my observations, the error is around machine precision.

To numerically match exactly,
```python
D = np.linalg.det(np.array([u, v]).transpose())
```
can be used, but this seems like overkill.

**Comment by mcara** [comment]
Why not import region from `gwcs`?

**Comment by WilliamJamieson** [comment]
> Why not import region from `gwcs`?

It already existed here. There are things in motion to refactor this into stcal as it is identical to gwcs and romancal implementations.

**Review by stscirij — APPROVED** [review]
This seems reasonable - it's not obvious that the determinant isn't the better way to go to see whether line segments are parallel

File: changes/8892.skymatch.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Resolve warnings emitted by NumPy 2 when running skymatch.
```
Comment (mcara) [inline changes/8892.skymatch.rst:1]: I wonder what is the policy for when change log entries are needed. This PR does not change API or provide any particular enhancement (to the user) or bug fix.

**Review by mcara — APPROVED** [review]

**Comment by mcara** [comment]
> There are things in motion to refactor this into stcal as it is identical to gwcs and romancal implementations.

I hope it is not going to `stcal`. I think it would be more helpful in `gwcs`. For example, `drizzlepac` also may use it (at least via `stsci.skypac`). Will it now need to import `stcal`? A better place, IMO, would be `gwcs` or `astropy`.

**Comment by mcara** [comment]
> There are things in motion to refactor this into stcal as it is identical to gwcs and romancal implementations.

I hope it is not going to `stcal`. I think it would be more helpful in `gwcs`. For example, `drizzlepac` also may use it (at least via `stsci.skypac`). Will it now need to import `stcal`? A better place, IMO, would be `gwcs` or `astropy`.

CC: @nden

**Comment by nden** [comment]
Regression tests show the same two failures as on main.
https://github.com/spacetelescope/RegressionTests/actions/runs/11449688800

**Review by nden — APPROVED** [review]

**Pull request merged**